Mathlib/Algebra/Category/ModuleCat/Presheaf/Sheafify.lean

```diff
@@ -317,4 +312,71 @@ noncomputable def sheafify : SheafOfModules.{v} R where
       map_smul := fun _ _ _ => by apply Sheafify.map_smul }
   isSheaf := A.cond
 
+/-- The canonical morphism from a presheaf of modules to its associated sheaf. -/
+@[simps]
+def toSheafify : M₀ ⟶ (restrictScalars α).obj (sheafify α φ).val where
+  hom := φ
+  map_smul X r₀ m₀ := by
+    simpa using (Sheafify.map_smul_eq α φ (α.app _ r₀) (φ.app _ m₀) (𝟙 _)
+      r₀ (by aesop) m₀ (by simp)).symm
+
+instance : Presheaf.IsLocallyInjective J (toSheafify α φ).hom :=
+  by dsimp; infer_instance
+
+instance : Presheaf.IsLocallySurjective J (toSheafify α φ).hom :=
+  by dsimp; infer_instance
+
+variable [J.WEqualsLocallyBijective AddCommGroupCat.{v}]
+
+/-- The bijection `((sheafify α φ).val ⟶ F) ≃ (M₀ ⟶ (restrictScalars α).obj F)` which
+is part of the universal property of the sheafification of the presheaf of modules `M₀`,
+when `F` is a presheaf of modules which is a sheaf. -/
+noncomputable def sheafifyHomEquiv' {F : PresheafOfModules.{v} R.val}
+    (hF : Presheaf.IsSheaf J F.presheaf) :
+    ((sheafify α φ).val ⟶ F) ≃ (M₀ ⟶ (restrictScalars α).obj F) :=
+  (restrictHomEquivOfIsLocallySurjective α hF).trans
+    (homEquivOfIsLocallyBijective (f := toSheafify α φ)
+      (N := (restrictScalars α).obj F) hF)
+
+lemma comp_sheafifyHomEquiv'_symm_hom {F : PresheafOfModules.{v} R.val}
+    (hF : Presheaf.IsSheaf J F.presheaf) (f : M₀ ⟶ (restrictScalars α).obj F) :
+    φ ≫ ((sheafifyHomEquiv' α φ hF).symm f).hom = f.hom :=
+  congr_arg Hom.hom ((sheafifyHomEquiv' α φ hF).apply_symm_apply f)
+
+/-- The bijection
+`(sheafify α φ ⟶ F) ≃ (M₀ ⟶ (restrictScalars α).obj ((SheafOfModules.forget _).obj F))`
+which is part of the universal property of the sheafification of the presheaf of modules `M₀`,
+for any sheaf of modules `F`. -/
```

Comment:
Can you reference the adjunction in the docstring?

Reply:
Thanks again for reviews! Done.